src/ORM/Table.php

```diff
@@ -2212,17 +2212,24 @@ public function hasFinder(string $type): bool
      * @return \Cake\ORM\Query
      * @throws \BadMethodCallException
      */
-    public function callFinder(string $type, Query $query, array $options = []): Query
+    public function callFinder(string $type, Query $query, ...$args): Query
     {
-        $query->applyOptions($options);
-        $options = $query->getOptions();
+        if ($args) {
+            $options = $args[0];
+            if (is_array($options)) {
```

Comment:
What about `ArrayAccess` things? Those were in the old doc string type information.

Reply:
`Query::applyOptions(array $options)` accepts array only. Looks like doc type info is wrong?

Reply:
Possible. We may have not ever actually had checks that were array specific before either.